File: frontend/types/work.ts
export interface Work {
  id: string;
  title: string;
  premise: string;
  genre: string;
  status_tag_id: string;
}

export interface CreateWorkInput {
  title: string;
  premise: string;
  genre: string;
  status_tag_id: string;
}

export interface UpdateWorkInput {
  title?: string;
  premise?: string;
  genre?: string;
  status_tag_id?: string;
}

File: backend/app/schemas/work.py
from pydantic import BaseModel
from typing import Optional


class WorkBase(BaseModel):
    title: str
    premise: str = ''
    genre: str = ''
    status_tag_id: str | None = None


class WorkCreate(WorkBase):
    pass


class WorkRead(WorkBase):
    id: str


class WorkUpdate(BaseModel):
    title: Optional[str] = None
    premise: Optional[str] = None
    genre: Optional[str] = None
    status_tag_id: Optional[str] = None
